### PR TITLE
local-pv-provisioner: Add PersistentVolumeReconciler to cleanup local PV

### PR DESCRIPTION
**Pull request by kfyharukz**
This PR implements Persistent Volume Reconciler to cleanup released local PV.
It's triggered by Update events of Persistent Volume and periodical trigger.

### CODE REVIEW

File: local-pv-provisioner/controllers/device_detector.go [hunk 1]
```diff
@@ -111,3 +128,55 @@ func (dd *DeviceDetector) do() error {
 
 	return nil
 }
+
+func (dd *DeviceDetector) pvName(devPath string) string {
```
Comment (kfyharukz) [inline local-pv-provisioner/controllers/device_detector.go:132]: The receiver methods of DeviceDetector are just moved from `persisntet_volume.go`.

File: local-pv-provisioner/controllers/device_detector_test.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
+package controllers
```
Comment (kfyharukz) [inline local-pv-provisioner/controllers/device_detector_test.go:1]: Just move the test for DeviceDetector to this file.

File: local-pv-provisioner/controllers/persistent_volume.go [hunk 3]
```diff
@@ -2,76 +2,176 @@ package controllers
...
-		log.Error(err, "unable to create or update PV", "device", dev)
 		return err
 	}
-	if op != controllerutil.OperationResultNone {
-		log.Info("PV successfully created or updated", "operation", op, "device", dev)
+	defer file.Close()
+
+	zeroBlock := make([]byte, d.FillBlockSize)
+	for i := uint(0); i < d.FillCount; i++ {
+		_, err = file.Write(zeroBlock)
```
Comment (ymmt2005) [inline local-pv-provisioner/controllers/persistent_volume.go:170]: Call `file.Sync` to persist the filling data.

Comment (ymmt2005) [reply local-pv-provisioner/controllers/persistent_volume.go:170]: Of course, after the loop.

File: local-pv-provisioner/controllers/persistent_volume.go [hunk 4]
```diff
@@ -2,76 +2,176 @@ package controllers
...
-	})
+// FillDeleter fills first 100MByte with '\0'
+type FillDeleter struct {
+	FillBlockSize uint
+	FillCount     uint
+}
+
+// Delete implements Deleter's method.
+func (d *FillDeleter) Delete(path string) error {
+	file, err := os.OpenFile(path, os.O_RDWR, 0)
```
Comment (ymmt2005) [inline local-pv-provisioner/controllers/persistent_volume.go:162]: Why do you use `os.O_RDWR`?

File: local-pv-provisioner/go.mod [hunk 5]
```diff
@@ -4,15 +4,17 @@ go 1.13
...
-	k8s.io/apimachinery v0.17.6
-	k8s.io/client-go v0.17.6
+	golang.org/x/crypto v0.0.0-20200221231518-2aa609cf4a9d // indirect
+	k8s.io/api v0.18.8
+	k8s.io/apimachinery v0.18.8
+	k8s.io/client-go v0.18.8
 	k8s.io/klog v1.0.0
-	sigs.k8s.io/controller-runtime v0.5.2
+	k8s.io/kube-openapi v0.0.0-20200410145947-bcb3869e6f29 // indirect
+	sigs.k8s.io/controller-runtime v0.6.2
```
Comment (ymmt2005) [inline local-pv-provisioner/go.mod:19]: Pleaes update this to v0.6.3

File: local-pv-provisioner/controllers/persistent_volume.go [hunk 6]
```diff
@@ -2,76 +2,177 @@ package controllers
...
+	defer file.Close()
+
+	zeroBlock := make([]byte, d.FillBlockSize)
+	for i := uint(0); i < d.FillCount; i++ {
+		_, err = file.Write(zeroBlock)
+		if err != nil {
+			return err
+		}
 	}
+	file.Sync()
```
Comment (ymmt2005) [inline local-pv-provisioner/controllers/persistent_volume.go:175]: You must check error.

**Review by ymmt2005 — CHANGES_REQUESTED** [review]
Please fix as commented.

**Review by ymmt2005 — APPROVED** [review]
LGTM

**Pull request merged**